### ISSUE TITLE
@PreferenceFunction annotation doesn't work with objects

### ISSUE THREAD

**Issue by thiagoms83** [skydoves/PreferenceRoom#20]
- Library Version: 1.2.0
- Affected Device(s): Within the code

Add a clear description about the problem.

@PreferenceFunction annotation doesn't work with target that is an object marked with @TypeConverter. Example:

```
@TypeConverter(BaseGsonConverter.class)
    protected AccessToken accessToken;

@PreferenceFunction("AccessToken")
    public AccessToken putAccessTokenFunction(AccessToken accessToken) {
        accessToken.update();
        return accessToken;
    }  
```
The code above generate a error code in the corresponding Preference_ file:
```
preference.edit().putString("AccessToken", super.putAccessTokenFunction(value.convertObject(accesstoken))).apply();
```

**Comment by skydoves** [comment]
Hi, @thiagoms83!
Could you let me know the error message or your example project?
Thanks!

**Comment by thiagoms83** [comment]
After run the task build in Preferences_Config in my example, it is display in messages:
```/home/marques/AndroidStudioProjects/croquidigital-android/domain/build/generated/ap_generated_sources/debug/out/pcdf/ic/domain/repository/local/Preference_Config.java:189: error: ')' expected
    return (AccessToken)super.getAccessTokenFunction(value.convertType(preference.getString("AccessToken", null));
                                                                                                                 ^
1 error
```

**Comment by skydoves** [comment]
Released a new stable `1.2.2`!
